grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -349,4 +343,13 @@ class PrepaymentInvoiceService {
             invoiceItem.invoice.addToInvoiceItems(inverseItem)
         }
     }
+
+    Integer getQuantityToInverse(OrderItem orderItem, Integer quantityInvoiced, Integer quantityInverseable) {
+        if (orderItem.isCompletelyFulfilled() && orderItem.isFullyInvoiced()) {
+            // If quantity if fully shipped and fully invoiced set full inverse quantity
+            return quantityInverseable
+        }
+
+        return quantityInvoiced >= quantityInverseable ? quantityInverseable : quantityInvoiced
+    }
```

Comment:
looks like a method that should be placed in the domain, what do you think @awalkowiak?

Reply:
I'd say it's difficult to determine in which domain it should be placed, because in most of those we operate on an order item, so it would be weird to place it in the `InvoiceItem` domain, especially that all the values passed in this method are not transients, but some values calculated in some other methods, e.g. `getQuantityToInverse` that is already placed in this service a few lines above.

I will let you guys debate on that and make a final decision

Reply:
Technically it potentially could be on the order item, but since it did not rely only on the data that could be pulled from the order item or prepayment item easily (it relies also on the specific invoice item quantity, currently created or edited) I think it is more suiting here.